charts/argo-cd/templates/argocd-application-controller/role.yaml

```
@@ -6,6 +6,9 @@ metadata:
   labels:
     {{- include "argo-cd.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
 rules:
+{{- if .Values.controller.roleRules.enabled }}
+  {{- toYaml .Values.controller.roleRules.rules | nindent 2 }}
+{{- else }}
```

We can remove `enabled` by using `with` but [the existing logic for clusterRoleRules](https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/values.yaml#L1135-L1141) has it... 🤕 Hmm

```suggestion
{{- .Values.controller.roleRules.with }}
  {{- toYaml . | nindent 2 }}
{{- end }}
```

---

I vote on using `with` since it's redundant to declare 2 attributes for users. 🙋 
*not aligned looks ugly of course.

---

Applied the with change as suggested!
I agree it’s cleaner and avoids redundancy. Thanks a lot for pointing this out! 🙌